File: Java/LiveData/app/src/main/java/com/example/livedataex/MainActivity.java
package com.example.livedataex;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

public class MainActivity extends AppCompatActivity {
    EditText nameInputView;
    TextView nameTextView;
    Button inputButton;

    NameViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        nameInputView = findViewById(R.id.inputName);
        nameTextView = findViewById(R.id.nameView);
        inputButton = findViewById(R.id.inputButton);

        viewModel = new ViewModelProvider(this , new NameViewModelFactory()).get(NameViewModel.class);

        Observer<String> nameObserver = newName -> nameTextView.setText(newName);

        viewModel.getCurrentName().observe(this, nameObserver);

        inputButton.setOnClickListener(view -> {
            String inputName = nameInputView.getText().toString();
            viewModel.getCurrentName().setValue(inputName);
        });
    }
}

File: Kotiln/SampleAppStudy/Kotlin_1/app/src/main/java/com/inik/kotlin_1/part0/chapter2/MainActivity.kt
package com.inik.kotlin_1.part0.chapter2

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.inik.kotlin_1.R

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //SAM
        //Single Abstract Method

        val view  = View(this)
        view.setOnClickListener({println("안녕")})
    }
}